pkg/cmd/util/clientcmd/factory.go

```diff
@@ -467,10 +468,34 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			if err != nil {
 				return nil, err
 			}
-			builds.Items = buildapi.FilterBuilds(builds.Items, buildapi.ByBuildConfigPredicate(t.Name))
-			if len(builds.Items) == 0 {
-				return nil, fmt.Errorf("no builds found for %q", t.Name)
+
+			hasConfigChangeTrigger := false
+			if t.Spec.Triggers != nil && len(t.Spec.Triggers) > 0 {
+				for _, trigger := range t.Spec.Triggers {
+					if trigger.Type == buildapi.ConfigChangeBuildTriggerType {
+						hasConfigChangeTrigger = true
+					}
+				}
+			}
+
+			err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+				builds.Items = buildapi.FilterBuilds(builds.Items, buildapi.ByBuildConfigPredicate(t.Name))
+				if len(builds.Items) > 0 {
+					return true, nil
+				} else {
+					if hasConfigChangeTrigger {
+						// Wait until timeout before giving up on the first build
+						return false, nil
+					} else {
+						return false, fmt.Errorf("no builds found for %q", t.Name)
```

Comment:
isn't this error going to get wrapped/eaten by line 495?  I think we want to retain the existing output behavior for cases where there was no build found and no configchangetrigger on the BC.